core/src/actors/connections_manager.rs

```diff
@@ -0,0 +1,179 @@
+use futures::Stream;
+use log::{debug, info};
+use std::net::SocketAddr;
+
+use actix::actors::resolver::{ConnectAddr, Resolver, ResolverError};
+use actix::fut::FutureResult;
+use actix::io::FramedWrite;
+use actix::{
+    Actor, ActorFuture, AsyncContext, Context, ContextFutureSpawner, Handler, MailboxError,
+    Message, StreamHandler, SystemService, WrapFuture,
+};
+use tokio::codec::FramedRead;
+use tokio::io::AsyncRead;
+use tokio::net::{TcpListener, TcpStream};
+
+use crate::actors::codec::P2PCodec;
+use crate::actors::session::Session;
+
+use witnet_p2p::sessions::SessionType;
+
+////////////////////////////////////////////////////////////////////////////////////////
+// ACTOR MESSAGES
+////////////////////////////////////////////////////////////////////////////////////////
+/// Actor message that holds the TCP stream from an inbound TCP connection
+#[derive(Message)]
+struct InboundTcpConnect {
+    stream: TcpStream,
+}
+
+impl InboundTcpConnect {
+    /// Method to create a new InboundTcpConnect message from a TCP stream
+    fn new(stream: TcpStream) -> InboundTcpConnect {
+        InboundTcpConnect { stream }
+    }
+}
+
+/// Actor message to request the creation of an outbound TCP connection to a peer.
+#[derive(Message)]
+pub struct OutboundTcpConnect {
+    /// Address of the outbound connection
+    pub address: SocketAddr,
+}
+
+/// Returned type by the Resolver actor for the ConnectAddr message
+type ResolverResult = Result<TcpStream, ResolverError>;
+
+////////////////////////////////////////////////////////////////////////////////////////
+// ACTOR BASIC STRUCTURE
+////////////////////////////////////////////////////////////////////////////////////////
+/// Connections manager actor
+#[derive(Default)]
+pub struct ConnectionsManager;
+
+/// Make actor from `ConnectionsManager`
+impl Actor for ConnectionsManager {
+    /// Every actor has to provide execution `Context` in which it can run.
+    type Context = Context<Self>;
+
+    /// Method to be executed when the actor is started
+    fn started(&mut self, ctx: &mut Self::Context) {
+        debug!("Connections Manager actor has been started!");
+
+        // Start server
+        // TODO[23-10-2018]: handle errors when starting server appropiately
```

Comment:
We should define a way to manage TODO/FIXME. I was following @csmoe suggestion but I think is a good idea open an issue to discuss it.

Reply:
the potential purpose of `// FIXME(#000):...` is forcing contributors to track everything with issues, thus we will have a clear workflow, which `TODO` will not.

Reply:
@mmartinbar changed most TODOs to FIXMEs following what we are saying.